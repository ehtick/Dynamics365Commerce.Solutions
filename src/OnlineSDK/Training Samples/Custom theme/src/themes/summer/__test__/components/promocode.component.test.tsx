/*!
 * Copyright (c) Microsoft Corporation.
 * All rights reserved. See LICENSE in the project root for license information.
 */

/* eslint-disable no-duplicate-imports */
import { ICartState } from '@msdyn365-commerce/global-state';
import { mount } from 'enzyme';
import * as React from 'react';

import PromoCode, { IPromoCodeProps } from '../../views/components/promocode.component';

const mockActionContext = {
    app: {
        config: {
            unitOfMeasureDisplayType: 'none'
        }
    },
    actionContext: {
        requestContext: {
            apiSettings: {
                baseImageUrl: 'https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/'
            }
        }
    },
    request: {
        gridSettings: {
            xs: { w: 767, h: 0 },
            sm: { w: 991, h: 0 },
            md: { w: 1199, h: 0 },
            lg: { w: 1599, h: 0 },
            xl: { w: 1600, h: 0 }
        },
        apiSettings: {
            channelId: 5637145359
        },
        channel: {
            PickupDeliveryModeCode: '70',
            channelDeliveryOptionConfig: 'channelDeliveryOptionConfig'
        }
    },
    cultureFormatter: {
        formatCurrency: jest.fn()
    }
};
const mockFunciton = jest.fn();

const mockActiveCart = {
    Id: 'gB8DCi6AK3mzMavh08C4P18wFDbEEyto',
    Version: 123,
    Coupons: ['a', 'b'],
    cart: {
        CartLines: [{ LineId: 123, DiscountLines: [{ OfferId: 123, DiscountCode: 'CodeXyz', DiscountCost: 10 }] }]
    }
};

const moduleProps: IPromoCodeProps = {
    cart: {
        totalItemsInCart: 3,
        isEmpty: false,

        // @ts-expect-error
        cart: mockActiveCart,
        CartLines: [{ LineId: 123 }],
        addPromoCode: jest.fn().mockImplementation(async () => Promise.resolve({ status: 'SUCCESS' })),
        removePromoCodes: jest.fn().mockImplementation(async () => Promise.resolve({ status: 'SUCCESS' }))
    },
    promoCodeHeadingText: '',

    // @ts-expect-error
    context: mockActionContext,
    appliedPromoCodeHeadingText: '',
    removePromoAriaLabelFormat: 'removePromoAriaLabelFormat',
    promoPlaceholderText: 'promoPlaceholderText',
    promoCodeApplyButtonText: 'promoCodeApplyButtonText',
    collapseTimeOut: 1,
    removePromoText: 'removePromoText',
    invalidPromoCodeErrorText: 'invalidPromoCodeErrorText',
    failedToAddPromoCodeErrorText: 'failedToAddPromoCodeErrorText',
    duplicatePromoCodeErrorText: 'duplicatePromoCodeErrorText ',
    failedToRemovePromoCodeErrorText: 'failedToRemovePromoCodeErrorText',
    promoCodeApplyCallback: mockFunciton
};

describe('Cartline', () => {
    it('renders correctly with change event', () => {
        const wrapper = mount(<PromoCode {...moduleProps} />);

        // Submit handler
        wrapper.find('.msc-promo-code__form-container').simulate('submit');
        wrapper.setProps({ removePromoAriaLabelFormat: false });
        wrapper.find('.msc-promo-code__input-box').simulate('change', { target: { value: 123 } });
        wrapper.find('button.msc-promo-code__apply-btn').simulate('click');

        wrapper.find('.msc-promo-code__input-box').simulate('change', { target: { value: '' } });
        wrapper.find('button.msc-promo-code__apply-btn').simulate('click');

        wrapper.setProps({ cart: undefined });
        wrapper.find('.msc-promo-code__input-box').simulate('change', { target: { value: 1234 } });
        wrapper.find('button.msc-promo-code__apply-btn').simulate('click');

        const cart: ICartState = {
            totalItemsInCart: 3,
            isEmpty: false,

            // @ts-expect-error
            cart: mockActiveCart,
            addPromoCode: jest.fn().mockImplementation(async () => Promise.resolve({ substatus: 'ALREADYADDED' }))
        };

        wrapper.setProps({ cart });
        wrapper.find('.msc-promo-code__input-box').simulate('change', { target: { value: 1234 } });
        wrapper.find('button.msc-promo-code__apply-btn').simulate('click');

        const cartFail: ICartState = {
            totalItemsInCart: 3,
            isEmpty: false,

            // @ts-expect-error
            cart: mockActiveCart,
            addPromoCode: jest.fn().mockImplementation(async () => Promise.resolve({ status: 'Fail' }))
        };

        wrapper.setProps({ cart: cartFail });
        wrapper.find('.msc-promo-code__input-box').simulate('change', { target: { value: 1234 } });
        wrapper.find('button.msc-promo-code__apply-btn').simulate('click');

        const cartFailError: ICartState = {
            totalItemsInCart: 3,
            isEmpty: false,

            // @ts-expect-error
            cart: mockActiveCart,
            addPromoCode: jest.fn().mockImplementation(async () => Promise.reject(new Error('Error')))
        };

        wrapper.setProps({ cart: cartFailError });
        wrapper.find('.msc-promo-code__input-box').simulate('change', { target: { value: 1234 } });
        wrapper.find('button.msc-promo-code__apply-btn').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly remove', () => {
        const wrapper = mount(<PromoCode {...moduleProps} />);
        wrapper
            .find('button.msc-promo-code__line__btn-remove')
            .at(1)
            .simulate('click');
        const instance = wrapper.instance() as PromoCode;
        const cartFail: ICartState = {
            totalItemsInCart: 3,
            isEmpty: false,

            // @ts-expect-error
            cart: mockActiveCart,
            addPromoCode: jest.fn().mockImplementation(async () => Promise.reject(new Error('Error'))),
            removePromoCodes: jest.fn().mockImplementation(async () => Promise.reject(new Error('Error')))
        };
        wrapper.setProps({ cart: cartFail });
        wrapper
            .find('button.msc-promo-code__line__btn-remove')
            .at(1)
            .simulate('click');

        // @ts-expect-error
        // eslint-disable-next-line  @typescript-eslint/no-confusing-void-expression -- have to export this as this utility is used in fabrikam tests
        const removePromotion = instance._removePromotion(undefined, {});
        expect(removePromotion).toBeUndefined();

        // @ts-expect-error
        // eslint-disable-next-line  @typescript-eslint/no-confusing-void-expression -- have to export this as this utility is used in fabrikam tests
        const removePromotionError = instance._removePromotion(cartFail, { currentTarget: { getAttribute: mockFunciton } });
        expect(removePromotionError).toBeUndefined();

        // @ts-expect-error
        const discountAmount = instance._calculateDiscount('CodeXyz', mockActiveCart);
        expect(discountAmount).toBe(-10);
    });
});
