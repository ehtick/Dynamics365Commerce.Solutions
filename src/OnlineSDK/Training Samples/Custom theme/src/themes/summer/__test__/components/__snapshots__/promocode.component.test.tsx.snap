// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cartline renders correctly with change event 1`] = `
<PromoCode
  appliedPromoCodeHeadingText=""
  cart={
    Object {
      "addPromoCode": [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "promoCode": 1234,
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Promise {},
          },
        ],
      },
      "cart": Object {
        "Coupons": Array [
          "a",
          "b",
        ],
        "Id": "gB8DCi6AK3mzMavh08C4P18wFDbEEyto",
        "Version": 123,
        "cart": Object {
          "CartLines": Array [
            Object {
              "DiscountLines": Array [
                Object {
                  "DiscountCode": "CodeXyz",
                  "DiscountCost": 10,
                  "OfferId": 123,
                },
              ],
              "LineId": 123,
            },
          ],
        },
      },
      "isEmpty": false,
      "totalItemsInCart": 3,
    }
  }
  collapseTimeOut={1}
  context={
    Object {
      "actionContext": Object {
        "requestContext": Object {
          "apiSettings": Object {
            "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
          },
        },
      },
      "app": Object {
        "config": Object {
          "unitOfMeasureDisplayType": "none",
        },
      },
      "cultureFormatter": Object {
        "formatCurrency": [MockFunction] {
          "calls": Array [
            Array [
              0,
            ],
            Array [
              0,
            ],
            Array [
              0,
            ],
            Array [
              0,
            ],
            Array [
              0,
            ],
            Array [
              0,
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      },
      "request": Object {
        "apiSettings": Object {
          "channelId": 5637145359,
        },
        "channel": Object {
          "PickupDeliveryModeCode": "70",
          "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
        },
        "gridSettings": Object {
          "lg": Object {
            "h": 0,
            "w": 1599,
          },
          "md": Object {
            "h": 0,
            "w": 1199,
          },
          "sm": Object {
            "h": 0,
            "w": 991,
          },
          "xl": Object {
            "h": 0,
            "w": 1600,
          },
          "xs": Object {
            "h": 0,
            "w": 767,
          },
        },
      },
    }
  }
  duplicatePromoCodeErrorText="duplicatePromoCodeErrorText "
  failedToAddPromoCodeErrorText="failedToAddPromoCodeErrorText"
  failedToRemovePromoCodeErrorText="failedToRemovePromoCodeErrorText"
  invalidPromoCodeErrorText="invalidPromoCodeErrorText"
  promoCodeApplyButtonText="promoCodeApplyButtonText"
  promoCodeApplyCallback={[MockFunction]}
  promoCodeHeadingText=""
  promoPlaceholderText="promoPlaceholderText"
  removePromoAriaLabelFormat={false}
  removePromoText="removePromoText"
>
  <div>
    <div
      className="msc-promo-code-heading"
    />
    <form
      className="msc-promo-code__form-container"
      onSubmit={[Function]}
    >
      <div
        className="msc-promo-code__group"
      >
        <input
          className="msc-promo-code__input-box"
          onChange={[Function]}
          placeholder="promoPlaceholderText"
          value={1234}
        />
        <Button
          className="msc-promo-code__apply-btn btn"
          disabled={false}
          onClick={[Function]}
          outline={false}
          tag="button"
          title="promoCodeApplyButtonText"
        >
          <button
            className="msc-promo-code__apply-btn btn msc-btn"
            disabled={false}
            onClick={[Function]}
            title="promoCodeApplyButtonText"
            type="button"
          >
            promoCodeApplyButtonText
          </button>
        </Button>
      </div>
    </form>
    <p
      aria-live="assertive"
      className=""
    />
    <div
      className="msc-promo-code__discount"
    >
      <div
        className="msc-promo-code__discount-heading"
      />
      <Component
        className="msc-promo-code__discount-value"
        context={
          Object {
            "actionContext": Object {
              "requestContext": Object {
                "apiSettings": Object {
                  "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
                },
              },
            },
            "app": Object {
              "config": Object {
                "unitOfMeasureDisplayType": "none",
              },
            },
            "cultureFormatter": Object {
              "formatCurrency": [MockFunction] {
                "calls": Array [
                  Array [
                    0,
                  ],
                  Array [
                    0,
                  ],
                  Array [
                    0,
                  ],
                  Array [
                    0,
                  ],
                  Array [
                    0,
                  ],
                  Array [
                    0,
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              },
            },
            "request": Object {
              "apiSettings": Object {
                "channelId": 5637145359,
              },
              "channel": Object {
                "PickupDeliveryModeCode": "70",
                "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
              },
              "gridSettings": Object {
                "lg": Object {
                  "h": 0,
                  "w": 1599,
                },
                "md": Object {
                  "h": 0,
                  "w": 1199,
                },
                "sm": Object {
                  "h": 0,
                  "w": 991,
                },
                "xl": Object {
                  "h": 0,
                  "w": 1600,
                },
                "xs": Object {
                  "h": 0,
                  "w": 767,
                },
              },
            },
          }
        }
        data={
          Object {
            "price": Object {
              "CustomerContextualPrice": 0,
            },
          }
        }
      >
        <Price
          className="msc-promo-code__discount-value"
          context={
            Object {
              "actionContext": Object {
                "requestContext": Object {
                  "apiSettings": Object {
                    "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
                  },
                },
              },
              "app": Object {
                "config": Object {
                  "unitOfMeasureDisplayType": "none",
                },
              },
              "cultureFormatter": Object {
                "formatCurrency": [MockFunction] {
                  "calls": Array [
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
              "request": Object {
                "apiSettings": Object {
                  "channelId": 5637145359,
                },
                "channel": Object {
                  "PickupDeliveryModeCode": "70",
                  "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
                },
                "gridSettings": Object {
                  "lg": Object {
                    "h": 0,
                    "w": 1599,
                  },
                  "md": Object {
                    "h": 0,
                    "w": 1199,
                  },
                  "sm": Object {
                    "h": 0,
                    "w": 991,
                  },
                  "xl": Object {
                    "h": 0,
                    "w": 1600,
                  },
                  "xs": Object {
                    "h": 0,
                    "w": 767,
                  },
                },
              },
            }
          }
          data={
            Object {
              "price": Object {
                "CustomerContextualPrice": 0,
              },
            }
          }
        >
          <span
            className="msc-price msc-promo-code__discount-value"
          >
            <span
              className="msc-price__actual"
              itemProp="price"
            />
          </span>
        </Price>
      </Component>
    </div>
    <div
      className="msc-promo-code__line-container"
    >
      <div
        className="msc-promo-code__line-value"
      >
         
        (
        <Component
          className="msc-promo-code__line-discount-value"
          context={
            Object {
              "actionContext": Object {
                "requestContext": Object {
                  "apiSettings": Object {
                    "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
                  },
                },
              },
              "app": Object {
                "config": Object {
                  "unitOfMeasureDisplayType": "none",
                },
              },
              "cultureFormatter": Object {
                "formatCurrency": [MockFunction] {
                  "calls": Array [
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
              "request": Object {
                "apiSettings": Object {
                  "channelId": 5637145359,
                },
                "channel": Object {
                  "PickupDeliveryModeCode": "70",
                  "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
                },
                "gridSettings": Object {
                  "lg": Object {
                    "h": 0,
                    "w": 1599,
                  },
                  "md": Object {
                    "h": 0,
                    "w": 1199,
                  },
                  "sm": Object {
                    "h": 0,
                    "w": 991,
                  },
                  "xl": Object {
                    "h": 0,
                    "w": 1600,
                  },
                  "xs": Object {
                    "h": 0,
                    "w": 767,
                  },
                },
              },
            }
          }
          data={
            Object {
              "price": Object {
                "CustomerContextualPrice": undefined,
              },
            }
          }
        >
          <Price
            className="msc-promo-code__line-discount-value"
            context={
              Object {
                "actionContext": Object {
                  "requestContext": Object {
                    "apiSettings": Object {
                      "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
                    },
                  },
                },
                "app": Object {
                  "config": Object {
                    "unitOfMeasureDisplayType": "none",
                  },
                },
                "cultureFormatter": Object {
                  "formatCurrency": [MockFunction] {
                    "calls": Array [
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  },
                },
                "request": Object {
                  "apiSettings": Object {
                    "channelId": 5637145359,
                  },
                  "channel": Object {
                    "PickupDeliveryModeCode": "70",
                    "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
                  },
                  "gridSettings": Object {
                    "lg": Object {
                      "h": 0,
                      "w": 1599,
                    },
                    "md": Object {
                      "h": 0,
                      "w": 1199,
                    },
                    "sm": Object {
                      "h": 0,
                      "w": 991,
                    },
                    "xl": Object {
                      "h": 0,
                      "w": 1600,
                    },
                    "xs": Object {
                      "h": 0,
                      "w": 767,
                    },
                  },
                },
              }
            }
            data={
              Object {
                "price": Object {
                  "CustomerContextualPrice": undefined,
                },
              }
            }
          />
        </Component>
        )
      </div>
      <Button
        aria-label=""
        className="msc-promo-code__line__btn-remove"
        onClick={[Function]}
        outline={false}
        tag="button"
        title="removePromoText"
      >
        <button
          aria-label=""
          className="msc-promo-code__line__btn-remove msc-btn"
          onClick={[Function]}
          title="removePromoText"
          type="button"
        >
          removePromoText
        </button>
      </Button>
    </div>
    <div
      className="msc-promo-code__line-container"
    >
      <div
        className="msc-promo-code__line-value"
      >
         
        (
        <Component
          className="msc-promo-code__line-discount-value"
          context={
            Object {
              "actionContext": Object {
                "requestContext": Object {
                  "apiSettings": Object {
                    "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
                  },
                },
              },
              "app": Object {
                "config": Object {
                  "unitOfMeasureDisplayType": "none",
                },
              },
              "cultureFormatter": Object {
                "formatCurrency": [MockFunction] {
                  "calls": Array [
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                    Array [
                      0,
                    ],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                },
              },
              "request": Object {
                "apiSettings": Object {
                  "channelId": 5637145359,
                },
                "channel": Object {
                  "PickupDeliveryModeCode": "70",
                  "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
                },
                "gridSettings": Object {
                  "lg": Object {
                    "h": 0,
                    "w": 1599,
                  },
                  "md": Object {
                    "h": 0,
                    "w": 1199,
                  },
                  "sm": Object {
                    "h": 0,
                    "w": 991,
                  },
                  "xl": Object {
                    "h": 0,
                    "w": 1600,
                  },
                  "xs": Object {
                    "h": 0,
                    "w": 767,
                  },
                },
              },
            }
          }
          data={
            Object {
              "price": Object {
                "CustomerContextualPrice": undefined,
              },
            }
          }
        >
          <Price
            className="msc-promo-code__line-discount-value"
            context={
              Object {
                "actionContext": Object {
                  "requestContext": Object {
                    "apiSettings": Object {
                      "baseImageUrl": "https://cms-ppe-imageresizer-mr.trafficmanager.net/cms/api/fabrikamsb/imageFileData/search?fileName=/",
                    },
                  },
                },
                "app": Object {
                  "config": Object {
                    "unitOfMeasureDisplayType": "none",
                  },
                },
                "cultureFormatter": Object {
                  "formatCurrency": [MockFunction] {
                    "calls": Array [
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                      Array [
                        0,
                      ],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  },
                },
                "request": Object {
                  "apiSettings": Object {
                    "channelId": 5637145359,
                  },
                  "channel": Object {
                    "PickupDeliveryModeCode": "70",
                    "channelDeliveryOptionConfig": "channelDeliveryOptionConfig",
                  },
                  "gridSettings": Object {
                    "lg": Object {
                      "h": 0,
                      "w": 1599,
                    },
                    "md": Object {
                      "h": 0,
                      "w": 1199,
                    },
                    "sm": Object {
                      "h": 0,
                      "w": 991,
                    },
                    "xl": Object {
                      "h": 0,
                      "w": 1600,
                    },
                    "xs": Object {
                      "h": 0,
                      "w": 767,
                    },
                  },
                },
              }
            }
            data={
              Object {
                "price": Object {
                  "CustomerContextualPrice": undefined,
                },
              }
            }
          />
        </Component>
        )
      </div>
      <Button
        aria-label=""
        className="msc-promo-code__line__btn-remove"
        onClick={[Function]}
        outline={false}
        tag="button"
        title="removePromoText"
      >
        <button
          aria-label=""
          className="msc-promo-code__line__btn-remove msc-btn"
          onClick={[Function]}
          title="removePromoText"
          type="button"
        >
          removePromoText
        </button>
      </Button>
    </div>
  </div>
</PromoCode>
`;
